### PR TITLE
Include owner and group into Unixfile metadata response and directory listing

### PR DESCRIPTION
**Pull request by lchudinov**
This PR includes `owner` and `group` fields into the Unixfile metadata and directory listing(JSON).

### CODE REVIEW

File: c/zosaccounts.c [hunk 1]
```diff
@@ -211,6 +308,11 @@ int userInfoGetUserId (UserInfo *info) {
   return userId;
 }
 
+/* Copy user name from User Info structure into userNameBuffer */
+void userInfoGetUserName (UserInfo *info, char *userNameBuffer) {
```
Comment (DivergentEuropeans) [inline c/zosaccounts.c:312]: Necessary to have its own method?

File: c/zosaccounts.c [hunk 2]
```diff
@@ -220,6 +322,11 @@ int groupInfoGetGroupId (GroupInfo *info) {
   return groupId;
 }
 
+/* Copy group name from Group Info structure into groupNameBuffer*/
+void groupInfoGetGroupName (GroupInfo *info, char *groupNameBuffer) {
```
Comment (DivergentEuropeans) [inline c/zosaccounts.c:326]: Necessary to have its own method?

File: c/zosaccounts.c [hunk 3]
```diff
@@ -211,6 +308,11 @@ int userInfoGetUserId (UserInfo *info) {
   return userId;
 }
 
+/* Copy user name from User Info structure into userNameBuffer */
+void userInfoGetUserName (UserInfo *info, char *userNameBuffer) {
+  memcpy(userNameBuffer, (const char*)&info->GIDN_U_NAME, info->GIDN_U_LEN);
```
Comment (DivergentEuropeans) [inline c/zosaccounts.c:313]: Necessary to have its own method?

File: c/zosaccounts.c [hunk 4]
```diff
@@ -266,6 +384,17 @@ int groupIdGet (char *string, int *returnCode, int *reasonCode) {
   return groupId;
 }
 
+/* Obtain group name by GID, groupNameBuffer must be at least 8 chars long */
+int groupGetName(int gid, char *groupNameBuffer, int *returnCode, int *reasonCode) {
+  int status = 0;
+  GroupInfo groupInfo = {0};
+  status = getGroupInfo(gid, &groupInfo, returnCode, reasonCode);
+  if (status == 0) {
```
Comment (DivergentEuropeans) [inline c/zosaccounts.c:392]: Should we check if groupNameBuffer is at least 8 chars long before getting here?

**Review by DivergentEuropeans — APPROVED** [review]
This works for me!

File: c/zosaccounts.c (hunk 2 above)
Comment (lchudinov) [reply c/zosaccounts.c:326]: It's just for the sake of consistency with the existing code that provides convenient access methods e.g. `groupInfo GetGroupId`

File: c/zosaccounts.c (hunk 3 above)
Comment (lchudinov) [reply c/zosaccounts.c:313]: It's just for the sake of consistency with the existing code that provides convenient access methods e.g. `userInfoGetUserId`

File: c/zosaccounts.c (hunk 4 above)
Comment (lchudinov) [reply c/zosaccounts.c:392]: We are unable to check if `groupNameBuffer` because the caller is responsible for allocating the buffer that is large enough.

**Review by 1000TurquoisePogs — APPROVED** [review]
Looks correct to me. I noticed the BPXY structs for user and group have username and groupname as a char and not a pointer, which I think could be improved, but this PR does not disturb existing code in that regard so it's good to merge.

**Comment by daveyc** [comment]
PMFJI, but the C structure mapping for `BPXYGIDS` and `BPXYGIDN` are not correct. The assembler DSECT `GIDS_G_NAME` is defined as `DS 0C` which has a zero duplication factor. This signifies that the the field has a variable length. To parse the results for the system calls will require pointer arithmetic using `GIDS_G_LEN`. This seems particularily brain-damaged as the maximum length of the field is only 8 bytes but every byte is sacred in HLASM land.

**Pull request merged**